packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -809,7 +805,18 @@ class StripeHelper {
         failure_message = /** @type {Charge} */ (charge).failure_message;
       }
 
-      const planName = await this.formatPlanDisplayName(sub.plan);
+      let product_id;
+      const product = sub.plan.product;
+
+      if (typeof product === 'string') {
```

Comment:
I recently added [an `expandResource` utility function](https://github.com/mozilla/fxa/blob/ff24a44895b582f0039e60ef02f42de0d3118958/packages/fxa-auth-server/lib/payments/stripe.js#L949) for Stripe fetches that does essentially this. Also, [an `expandAbbrevProductForPlan` utility](https://github.com/mozilla/fxa/blob/ff24a44895b582f0039e60ef02f42de0d3118958/packages/fxa-auth-server/lib/payments/stripe.js#L962) to get Product details out of cache (i.e. id, name, and metadata). One of those might be handy here